### PR TITLE
Move and Rename commands

### PR DESCRIPTION
**Pull request by trabus**
[rendered](https://github.com/trabus/rfcs/blob/move-rename/active/0000-move-rename.md)

### CODE REVIEW

File: active/0000-move-rename.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+Because the mv command has a long established history, the limited functionality of this implementation could cause confusion for many users. There also may be many corner cases that won't immediately be covered by the command, possibly breaking apps, causing more issues than it solves. As long as the command leaves a trail and log of what it did, undoing any accidental changes should be managable.
+
+If the mv command is used to update several files in sequence, it will not be very efficient, as it will be repeating the same task several times without the ability to consolidate. However, performance is less of a concern in regard to migration/refactoring, as the user shouldn't need to do so frequently.
+
+# Alternatives
+
+Ember-cli alternatives are unknown, the only way to do this currently is manually through the command line and with a text editor with search and replace capabilities.
+
+# Unresolved questions
+
```
Comment (stefanpenner) [inline active/0000-move-rename.md:174]: - does this rename components in `HBS` files?
- does this rename/move route information in config/router.js?

**Comment by stefanpenner** [comment]
lots of good detail here, but it seems like we may need to explore in-more detail how various transformations (in-file) occur.

Also, where are transform operations defined? We also don't want each transform to re-invent all source-transforms, some interopt with route-generator etc. will likely need to exist.

**Comment by trabus** [comment]
> lots of good detail here, but it seems like we may need to explore in-more detail how various transformations (in-file) occur.
> 
> Also, where are transform operations defined? We also don't want each transform to re-invent all source-transforms, some interopt with route-generator etc. will likely need to exist.

My initial work on `ember-cli-mv` stalled because of exactly those questions. I ran into some dangerous/destructive situations that I hadn't accounted for in the initial spec. I will need to revisit this soon with these questions in mind.

**Comment by stefanpenner** [comment]
cc @ladyleet

**Comment by ladyleet** [comment]
@stefanpenner 👍

**Comment by jbandura** [comment]
hi, just wanted to ask if this is still under consideration? From my side it looks super helpful and could make developers life easier (especially renaming/moving the components tests, templates and js file).

**Comment by Turbo87** [comment]
closing due to inactivity

once MU lands this will become a lot easier since we only have to move/rename folders then

**Pull request closed without merging**